presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -243,18 +243,20 @@ public void testFloor()
         assertFunction("floor(-123.0)", DOUBLE, -123.0);
         assertFunction("floor(123.45)", DOUBLE, 123.0);
         assertFunction("floor(-123.45)", DOUBLE, -124.0);
-        assertFunction("floor(FLOAT '123.0')", DOUBLE, (double) 123.0f);
-        assertFunction("floor(FLOAT '-123.0')", DOUBLE, (double) -123.0f);
-        assertFunction("floor(FLOAT '123.45')", DOUBLE, (double) 123.0f);
-        assertFunction("floor(FLOAT '-123.45')", DOUBLE, (double) -124.0f);
+        assertFunction("floor(FLOAT '123.0')", FLOAT, 123.0f);
+        assertFunction("floor(FLOAT '-123.0')", FLOAT, -123.0f);
+        assertFunction("floor(FLOAT '123.45')", FLOAT, 123.0f);
+        assertFunction("floor(FLOAT '-123.45')", FLOAT, -124.0f);
         assertFunction("floor(CAST(NULL as DOUBLE))", DOUBLE, null);
+        assertFunction("floor(CAST(NULL as FLOAT))", FLOAT, null);
     }
 
     @Test
     public void testLn()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("ln(" + doubleValue + ")", DOUBLE, Math.log(doubleValue));
+            assertFunction("ln(FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.log((float) doubleValue));
```

Comment:
Don't we have `float ln(float)`? We should expected `FLOAT` to be returned here, don't we?